File: CodinGame.Application/Models/CalculateWeightRequest.cs
using CodinGame.MarsLander.Models;
using CodinGame.MarsLander.Models.Dtos;

namespace CodinGame.Application.Models
{
    public class CalculateWeightRequest
    {
        public EvolutionParameters Parameters { get; set; }
        public Map Map { get; set; }
        public AiWeight OriginalWeights { get; set; }
        // Change to be tried up and down.
        public double ChangePerTry { get; set; }
        // Overall amount of tries into each direction.
        public int UpwardTries { get; set; }
        public int DownwardTries { get; set; }
    }
}

File: CodinGame.Application/client/src/views/mars-lander/api/requests/CalculateWeightRequest.ts
import {EvolutionParameters} from "@/views/mars-lander/models/evolution/EvolutionParameters";
import {AiWeight} from "@/views/mars-lander/models/evolution/AiWeight";
import {Map} from "@/views/mars-lander/models/environment/Map";

export class CalculateWeightRequest {
  public Parameters!: EvolutionParameters;
  public Map!: Map;
  public OriginalWeights!: AiWeight;
  public ChangePerTry!: number;
  public UpwardTries!: number;
  public DownwardTries!: number;

  constructor(init?: Partial<CalculateWeightRequest>) {
    Object.assign(this, init);
  }
}